docs: added short explanation about event reply

Hyron48 opened the pull request:
a short explanation about event reply has been added, briefly explaining the phases that compose it

## PR Checklist
Please check if your PR fulfills the following requirements:

- [X] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/contributing-docs/commit-message-guidelines.md
- [X] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [X] Documentation content changes
- [ ] angular.dev application / infrastructure changes
- [ ] Other... Please describe:


## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->

Issue Number: #60115 

## What is the new behavior?

a short explanation about event reply has been added, briefly explaining the phases that compose it


## Does this PR introduce a breaking change?

- [ ] Yes
- [X] No


<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->

thePunderWoman commented on adev/src/content/guide/hydration.md:
```diff
@@ -75,6 +75,24 @@ bootstrapApplication(App, {
+During the **Hydration** phase, Event Replay intercepts and stores all interactions that the user may perform, such as clicks and other events.
+
+- **Storing events**<br>
+The **Event Emitter** keeps in memory all the interactions recorded in the previous step, ensuring that they are not lost during the Hydration process.
```
```suggestion
The **Event Contract** keeps in memory all the interactions recorded in the previous step, ensuring that they are not lost for later replay.
```

Event replay actually happens post-hydration since we need the components to fully initialize before we can safely replay them.

thePunderWoman commented on adev/src/content/guide/hydration.md:
```diff
@@ -75,6 +75,24 @@ bootstrapApplication(App, {
+The **Event Emitter** keeps in memory all the interactions recorded in the previous step, ensuring that they are not lost during the Hydration process.
+
+- **Relaunch of events**<br>
+Once **Hydration** is complete, Angular rethrows the captured events, updating the application state and ensuring a smooth transition without losing interactions.
```
```suggestion
Once **Hydration** is complete, Angular re-invokes the captured events, ensuring none of those user actions were lost.
```

thePunderWoman commented on adev/src/content/guide/hydration.md:
```diff
@@ -75,6 +75,24 @@ bootstrapApplication(App, {
+
+---
+
+This feature ensures a consistent user experience, preventing actions performed before Hydration from being ignored.
```
```suggestion
This feature ensures a consistent user experience, preventing user actions performed before Hydration from being ignored.
```

thePunderWoman reviewed (changes requested):
There's a few language issues here that need some tweaking, but otherwise it's a good start for us. Thanks for your effort here.

Hyron48 committed:
docs: added short explanation about event reply

a short explanation about event reply has been added, briefly explaining the phases that compose it

Hyron48 commented:
@thePunderWoman thank you so much for the feedback. I just updated the commit with the requested changes. I hope they are ok

thePunderWoman reviewed (approved):
LGTM! Thank you for the fixes!

pkozlowski-opensource commented:
This PR was merged into the repository by commit 55ba1ededdbd0799830ae37048ac4402bc0dc2b2.

The changes were merged into the following branches: main, 19.2.x
